beacon_node/beacon_chain/src/block_verification_types.rs

```diff
@@ -478,12 +494,13 @@ impl<E: EthSpec> AsBlock<E> for AvailableBlock<E> {
     }
 
     fn into_rpc_block(self) -> RpcBlock<E> {
-        let (block_root, block, blobs_opt) = self.deconstruct();
+        let (block_root, block, blobs_opt, data_columns_opt) = self.deconstruct();
         // Circumvent the constructor here, because an Available block will have already had
         // consistency checks performed.
-        let inner = match blobs_opt {
-            None => RpcBlockInner::Block(block),
-            Some(blobs) => RpcBlockInner::BlockAndBlobs(block, blobs),
+        let inner = match (blobs_opt, data_columns_opt) {
+            (None, None) => RpcBlockInner::Block(block),
+            (Some(blobs), _) => RpcBlockInner::BlockAndBlobs(block, blobs),
+            (_, Some(data_columns)) => RpcBlockInner::BlockAndDataColumns(block, data_columns),
```

Comment:
What happens here if we have both blobs and data_columns available? I guess we'd return a `BlockAndBlobs` variant? Would that imply that if blobs are avail, we'll just skip the data sampling checks?

I might be totally off base here, but would a `BlockAndBlobsAndDataColumns` variant allow us to support both `blobs` and `data_columns` until were ready to migrate away from blob subnets completely?

Reply:
Yeah, so originally I was thinking to prioritise blobs, but then it's kind of pointless because we aren't executing the data column code paths.

I've been thinking about this and talking to Sean about this as well. A potentially way to introduce this and test it out may be add column subnets and sampling (without making it mandatory for `is_data_avaialble`, perhaps log warnings if it fails), and keep the existing blob subnets and DA filter logic - so perhaps an intermediate `BlockAndBlobsAndDataColumns` variant may make sense until we fully switch over to 1D PeerDAS.

Reply:
Can you 
```
match self {
  RpcBlockInner::*=> RpcBlock
}
```
so the enum type guarantees that blobs and columns do not exist at the same time

Reply:
Not sure if I'm understanding correctly but `self` is actually a struct type which contains optional blobs and columns:

https://github.com/sigp/lighthouse/blob/e707536fd5468c58493b5185e6566cee9bf9d392/beacon_node/beacon_chain/src/data_availability_checker.rs#L603-L608

Reply:
For an `RpcBlock`, we will have either `block + blobs` or `block + samples` if we've transitioned to DAS.